pmg/views.py

```diff
@@ -1701,20 +1701,59 @@ def correct_this_page():
 @app.route('/blog/<int:page>/')
 def blog(page=0):
     per_page = 10
+    filters = {}
+
+    year = func.date_part('year', Post.date).label('year')
+    month = func.date_part('month', Post.date).label('month')
+    month_name = func.to_char(Post.date, 'FMMonth').label('month_name')
+
+    post_query = Post.query
+
+    if request.args.get('filter[month]'):
+        filters['month'] = request.args.get('filter[month]')
+        post_query = post_query.filter(month_name == filters['month'])
 
-    count = Post.query.count()
-    posts =  Post.query\
+    if request.args.get('filter[year]'):
+        filters['year'] = request.args.get('filter[year]')
+        post_query = post_query.filter(year == filters['year'])
+
+    count = post_query.count()
+    posts =  post_query\
                 .order_by(desc(Post.date))\
                 .limit(per_page)\
                 .offset(page*per_page)\
                 .all()
 
+    months = db.session.query(
+            month_name, 
+            month, 
+            year, 
+            func.count(Post.id).label('month_posts')
+        )\
+        .group_by('year', 'month', 'month_name')\
+        .order_by(year.desc(), month.asc())\
+        .subquery('months')
+
+    years = db.session.query(
+                cast(months.c.year, Integer).label('year'), 
+                func.sum(months.c.month_posts).label('year_posts'), 
+                func.array_agg(months.c.month).label('months'), 
+                func.array_agg(months.c.month_name).label('month_names'), 
+                func.array_agg(months.c.month_posts).label('months_posts'))\
+        .group_by('year')\
+        .order_by(year.desc())\
+        .all()
```

Comment:
Returns something like:
```
[{
 'month_names': [u'May'], 
 'months': [5.0], 
 'months_posts': [1L],
 'year_posts': Decimal('1'),
 'year': 2023
}, {
 'month_names': [u'May', u'September', u'August', u'January'], 
 'months': [5.0, 9.0, 8.0, 1.0],
 'months_posts': [1L, 1L, 12L, 1L],
 'year_posts': Decimal('15'),
 'year': 2019
}]
```

Where each index in `month_names`, `months` and `months_posts` corresponds to each other. I.e. `month_names[0]` is the month's name for `month[0]` and `months_posts[0]` is the number of posts in `month[0]`.

I could figure out how to rather return a single array for those three properties combined instead, e.g. `[{ 'month_name': 'May', 'month': 5, 'months_posts': 1}, ...]`.